src/reducers/index.js

```diff
@@ -17,17 +17,28 @@ import IntegerBufferSet from '../vendor_upstream/struct/IntegerBufferSet';
 import PrefixIntervalTree from '../vendor_upstream/struct/PrefixIntervalTree';
 import shallowEqual from '../vendor_upstream/core/shallowEqual';
 
-import columnTemplates from '../selectors/columnTemplates';
 import convertColumnElementsToData from '../helper/convertColumnElementsToData';
 import { getScrollAnchor, scrollTo } from './scrollAnchor';
 import { getColumnAnchor, scrollToX as scrollToXAnchor } from './columnAnchor';
-import columnStateHelper from './columnStateHelper';
 import computeRenderedRows from './computeRenderedRows';
 import Scrollbar from '../plugins/Scrollbar';
 import { createSlice } from '@reduxjs/toolkit';
 import computeRenderedCols from './computeRenderedCols';
+import { initializeFlexColumnWidths } from './flexColumnWidths';
 import clamp from 'lodash/clamp';
 
+// NOTE (pradeep): Custom class objects are ignored by immer. (see https://immerjs.github.io/immer/complex-objects/)
+// We wrap our non-user defined structures from "internal" state into these wrapper classes so that immer
+// will skip it entirely while drafting the state.
+// An example is `storedHeights`, which is a simple array of row heights typically containing millions of entries.
+// Having this drafted by immer for every scroll reduces performance.
+function ArrayWrapper(...args) {
+  this.array = new Array(...args);
+}
+function ObjectWrapper(...args) {
+  this.object = new Object(...args);
+}
```

Comment:
I basically did this because I was seeing performance issues when modifying arrays of 1000000 entries.
Objects like `storedHeights` are deemed as "internal" state. These are only used in this file, are mutated in place, can't be used for redux history, and also not passed down to the components. So it should be a perfect candidate to be skipped by immer.
cc: @deshpsuy